### PR TITLE
Add DNA codon

### PR DESCRIPTION
**Pull request by SkeletalDemise**
Adds ability to decode DNA codons into the files. Requires a dictionary/table that should be added to CipheyDists. Major problem is that because there are no spaces in between the words when decoded, Ciphey doesn't consider it to be English, and doesn't find it. This is the same issue with the A1Z26 cipher. 

This is going to have to be solved with a different type of checker. Maybe hold off on merging this for a bit?

### CODE REVIEW

File: ciphey/basemods/Decoders/dna.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        """
+        Performs DNA decoding
+        """
+        logger.trace("Attempting DNA decoder")
+        ctext_decoded = ""
+        ctext = re.sub(r"[,;:\-\s]", "", ctext)
+        ctext = " ".join(ctext[i : i + 3] for i in range(0, len(ctext), 3))
+        ctext_split = ctext.split(" ")
+
+        try:
```
Comment (bee-san) [inline ciphey/basemods/Decoders/dna.py:22]: I don't like this try except :( 

You can check if a string exists in a dictionary with:

```python
x = {"hello": 6}
keys = x.keys()
if "food" in keys:
```

Other than that, it's good! :)

**Review by bee-san — CHANGES_REQUESTED** [review]

**Comment by bee-san** [comment]
Is this turned off? so if I pull it, it won't effect ciphey until we turn it on?

File: ciphey/basemods/Decoders/dna.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            for i in ctext_split:
+                ctext_decoded += self.DNA_DICT[i]
+            logger.debug(f"DNA decoded string is {ctext_decoded}")
+            return ctext_decoded
+        except Exception:
+            return None
+
+    @staticmethod
+    def priority() -> float:
+        return 0.2
```
Comment (bee-san) [inline ciphey/basemods/Decoders/dna.py:32]: 0.2 is very very high, I have never seen DNA in a CTF before :P

Comment (SkeletalDemise) [reply ciphey/basemods/Decoders/dna.py:32]: Yeah I know. I think the reason why I did that, was it wasn't working when I put it lower. Honestly this whole priority system isn't that good in my opinion.

File: ciphey/basemods/Decoders/dna.py (hunk 1 above)
Comment (SkeletalDemise) [reply ciphey/basemods/Decoders/dna.py]: Okay, I removed the try except. I tested it, it works good.

**Comment by SkeletalDemise** [comment]
Yes, it's turned off. I didn't add it to init.py

File: ciphey/basemods/Decoders/dna.py (hunk 2 above)
Comment (bee-san) [reply ciphey/basemods/Decoders/dna.py:33]: Yup! It sucks, it'll be fixed in the next searcher update. Right after tagging and new checkers :sweat_smile:

**Comment by bee-san** [comment]
Merging this because if it's turned off, we'll just turn it back on with the addition of new checkers :)

**Pull request merged**